### PR TITLE
Overhauls how admin prison incarcerates mobs

### LINKED ISSUE #60967: Admin prisoned players with Telekinesis can access visible consoles
## Reproduction:

If a player with Telekinesis is put into the admin prison they can interact with various consoles in their view range, one of these includes a security record console which they can use to alter security records and/or purge them, which is something we ideally don't want admin prisoned players doing

### PR DESCRIPTION
**Pull request by ZephyrTFA**
See title

Players in admin prison will now have a var on their mob, `admin_input_block_enabled ` set to `TRUE` which will cause Click, DblClick, and keyLoop to all return and not do anything; preventing them from doing stuff such as use tk on consoles in view

Closes https://github.com/tgstation/tgstation/issues/60967

:cl:
admin: Players in jail are no longer allowed to do anything other than respond to their ahelp
admin: After resolving their ticket, you must use the new player panel button 'Release from Prison' to release them.
/:cl:

### CODE REVIEW

**Comment by NamelessFairy** [comment]
Adding in that this should fix #60967

File: code/modules/admin/prison.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+/mob/forceMove(atom/destination)
+	if(admin_input_block_enabled)
+		return
+	return ..()
+
+/mob/abstract_move(atom/destination)
+	if(admin_input_block_enabled)
+		return
+	return ..()
```
Comment (Watermelon914) [inline code/modules/admin/prison.dm:83]: these are some of the hottest procs, you are overriding them which'll incur proccall overhead. Please look into a different way of doing this

**Review by Watermelon914 — DISMISSED** [review]

**Comment by optimumtact** [comment]
I think it's highly likely admins will forget to release people, anyway you can register for an obj exit signal on the area and then unapply the changes automatically?

Also, this adds an if check to a bunch of hot procs for a single purpose which can easily be resolved by simply moving any consoles and machines out of admin prisoned users views.

not sure about this PR tbh

File: code/modules/admin/prison.dm [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/// The admin prison watcher datum this mob has been assigned
+/mob/var/datum/admin_prison_watcher/admin_prison_holder
```
Comment (Mothblocks) [inline code/modules/admin/prison.dm:2]: This isn't Skyrat, don't do it like this because it messes up SpacemanDMM

**Comment by Mothblocks** [comment]
I agree that this shouldn't be handled through Click

**Pull request closed without merging**